### DISCUSSION TITLE
GSW141

### DISCUSSION THREAD

**Discussion by sethp** [rustbox/discussion#14] [General]
An (editable) placeholder to collect some discussion about the part in the title.

**Comment by sethp** [comment]
# Eval Kit

There's a $300 [eval board](https://www.mouser.com/ProductDetail/MaxLinear/GSW145-EVK-1?qs=T%252BzbugeAwjiD%252BoxwMecrMQ%3D%3D) for the GSW145 

(through the highly advanced technique of "flipping back and forth really fast" I've compared the GSW145 to the GSW141; I think the main/only difference is that the former is capable of 2.5G speeds, so I suspect they're actually the same design that's just been binned differently)
 
The best reference I've found so far for it is the photograph by way of: https://www.mouser.com/new/maxlinear/maxlinear-ethernet-eval-boards

![EasyGSW140A3 image; verbal description follows](https://github.com/rustbox/discussion/assets/241129/c2bea10a-7508-450b-815e-2b5269dfc364)

In roughly left-right, top-bottom order, I see:

1. A row of ports that "open" along the top, some labels
    * Leftmost is an SFP port labeled "2.5G/SGMII" (the silkscreen says "CON2")
    * Then, what appears to be 4x RJ-45 ports labeled with:
        <table>
             <tr><td colspan="3">GSW140 Ref. Board</td></tr>
             <tr><td>A31</td><td>Default Assembly</td><td>V1.2.2</td></tr>
             <tr><td>1746</td><td>SNR 003</td><td>P00</td></tr>
        </table>
    * Then, a single RJ-45 port with a QR code that decodes to "EASY7085(GSW140)_Ref_v122_A31_V1.3_TTT_1746SNR003"
    * Rightmost is an SFP port with what appears to be a quality control sticker on it (the silkscreen says "CON3")
2.  A row of magnetics (transformers)
    * 2x of what seem like the same two-port part but with different labels (1650 vs 1651) but the same part number? (HN4843CG)
    * A single-port-sized transformer labeled "1651 / HN2443CG" (where / indicates a line break)
3. An IC with what looks like the hand-written initials "SC" on it (no part number visible) with the silkscreen "U1" (no leads are visible)
4. An IC with part number PEF7072 (a single-port PHY transceiver) 
    * There's an inductor L3 nearby that's labeled 4R7 (4.7µH?)
5. A USB-B port of some kind on the rightmost edge silk-screened CON1
6. A barrel jack connector with a red sticker that says "+5V" and is silkscreened DCIN +5V
7. Two crystals ([corroboration](https://electronics.stackexchange.com/a/16796/174014))
    * One silkscreened Y1 next to "U1", no visible part number
    * One silkscreened Y2 next to "U8", reads (roughly) "TXC o/ 25.0Uc3G" (where / indicates a line break)
8. What looks like a power regulation circuit driven by IC "U2" (no part number visible) with some inductors (L1/L2), capacitors, a pair of (0Ω?) resistors
9. lots and lots of very tiny misc. resistors and capacitors
    * though, almost all of them look like they're unpopulated? especially the ones nearby "U8"
10. Three (unpopulated) headers
    * J3 looks like JTAG for "U1"
    * J4 looks like JTAG for "U8"
    * J15 breaks out 8 pins from roughly the "southwest" corner of "U1"
11. A totally unfamiliar part I don't recognize labeled "U3" (right next to "J3")
12. There's lots of "stippling" or "raised dots" or "braille" on the left "half" of the board, especially underneath/around a the big "TTT" sticker 
13. Around "U1" there's also lots of raised solder bumps in between pairs of traces

**Reply by sethp** [reply]
I think "U1" is the GSW145 chip, and "U8" is just there to interface between RGMII and the second SFP port (at a guess: SFP ports are meant to speak SGMII).

I think that means a stripped down version of this could be as simple as 1-3 (omitting the SFP ports, natch) with some sort of power network, a single crystal, and a single JTAG header.

With the right pin-strapping, that alone seems like it ought to be able to act as a fairly intelligent switch (albeit one whose configuration cycles runs through kicad re-design and PCB manufacture). 

Some open questions for me:

- [ ] What is "U3"? Is that a connector on top of it?
- [ ] Why are there all these unpopulated capacitor/resistor slots?
- [ ] What the heck does the USB port connect to? It just purely for power?
- [ ] What are the details of the power supply network? (AKA how do we build one?)
- [ ] What is the "stippling" from 12 called? What does it do (presumably, some sort of filtering/analog noise reduction)?
- [ ] The raised solder bumps from 13 are presumably between differential pairs; how do they provide "cross-talk" shielding? By adding capacitance (i.e. connecting to ground) or inductance?
- [ ] It looks like there's one (populated) capacitor per differential pair next to the magnetics; what that for?

**Reply by sethp** [reply]
Oh, and one more:

- [ ] How do we get the xMII signals "out" for development purposes? 

Seems like SGMII would be a lot easier to wrangle, being something like 2-6 pins as opposed to RGMII's, like 10-15 or whatever. The cost there being a higher clock rate, though, which means we'd need some care in the connector: maybe an edge connector? Or some other very short board->board trace?

**Reply by sethp** [reply]
re:

>  What is the "stippling" from 12 called? What does it do (presumably, some sort of filtering/analog noise reduction)?

Per @dougli1sqrd that's probably a bunch of vias that tie the ground planes together, a technique known as "via stitching"

**Reply by dougli1sqrd** [reply]
The fact that the solder (or whatever it is) is in between the signal pairs could indicate ... something? But if it's acting as a "barrier" between different pairs of signals that's what that ground pour between the signals should be for anyway. I tried googling around but didn't really find anything. I have like medium confidence that those aren't "needed"? I feel like it must be more than just solder, otherwise you wouldn't do it like that. If you wanted to put solder on the board you have to expose copper there on the pcb. But why expose copper just to cover it with solder? The solder doesn't seem like it would do anything electrically more than just having copper there. So I can only guess that it's soldering _something_ there. But what I have no idea.

**Reply by dougli1sqrd** [reply]
And yeah I'm thinking edge connector, and also test points to probe what's going out of the xMII interfaces

**Reply by sethp** [reply]
> The solder doesn't seem like it would do anything electrically more than just having copper there. So I can only guess that it's soldering something there. But what I have no idea.

Maybe it's connected to the "shield" bits, rather than ground? Or, more conductive material implies more inductive "inertia", right? And perhaps it's the 3D-ness of it is what makes it superior to a "flat" copper trace in terms of, like, "attracting" the noisy emissions from the differential pair? (isn't one of the benefits of differential pairs that the noise emissions kinda cancel each other out tho?)

I appreciate you looking around, either way. It's definitely very strange!

**Comment by dougli1sqrd** [comment]
In my preliminary PCB routing playing around, I've been using the https://www.mouser.com/ProductDetail/ABRACON/ARJM11C7-502-KB-EW2?qs=YCa%2FAAYMW00En29mj%252BAmyw%3D%3D as the ethernet RJ45 jacks as they have internal magnetics and cheap and available.

I am curious though if this is the part, or if I have the pinouts correct? Like, It's kinda fine, but it seems like each rj45 jack _and_ each diff pair within each jack are all swapped so that everything has to be routed around everything else.
![Screenshot from 2024-06-30 19-03-09](https://github.com/rustbox/discussion/assets/1422171/aa58eb81-a1ad-4bfa-9b8c-02590f49d4b8)

It'd be much easier if the individual diff pairs and the order of the diff pairs were flipped the lines could come directly out of the RJ45 pins into the Switch chip more like in this image
![Screenshot from 2024-06-30 19-07-48](https://github.com/rustbox/discussion/assets/1422171/afb1aa68-6932-4cfc-99d5-cbe0f5d479bc)

Not sure if a different part would have a different pinout that would be easier to route?

**Reply by dougli1sqrd** [reply]
Okay I've discovered that you can select "tab up" vs "tab down" RJ45 jacks. "Tab down" refers to the following image:
![image](https://github.com/rustbox/discussion/assets/1422171/ec2ee9e3-9b99-47af-8eb8-0c10ef3e06b4)
Where the flat part with the pins of the cable goes up and the "tab" faces downward. This is the one that produces the untwisted arrangement as shown in the last picture in the main body.

This is "tab up":
![image](https://github.com/rustbox/discussion/assets/1422171/b61d20bd-5b09-454b-b95a-84ac1e65c9d3)
Which is what I was accidentally using in my initial PCB playing. 

I think for our purposes, routing traces for this GSW141 will be easiest if we use a "Tab Down" jack.

**Reply by dougli1sqrd** [reply]
Another question I have is what difference the Base 10/100 T jacks vs the base 1000 T jacks are. Like there are the right number of pins present (8 for the 4 twisted pairs) and they each went through magnetics is there a real difference in functionality? Or is having 8 pins the same as saying it's for Base 1000 T?

(is it 1000 Base T?)

**Reply by dougli1sqrd** [reply]
So based on the filters, there aren't many options to select for RJ45 especially at a nice price point. The example board uses https://www.lcsc.com/product-detail/Ethernet-Connectors-Modular-Connectors-RJ45-RJ11_UDE-Corp-RB1-125B8G1A_C363353.html which has pretty steep shipping costs but with the cheapest per unit RJ45. The advantage here is that the symbol and footprint already exist in kicad. I may try playing with this symbol / footprint for the time being just to make a design.

**Reply by sethp** [reply]
The RJ-45 jacks are through hole parts, right? So isn't the difference between the tab up and tab down jacks the same as mounting the part "upside down" on the board (looks like kicad calls this "flip")? Not that I'm being paid off by big "tab up" or anything, just checking if my understanding is right.

**Reply by sethp** [reply]
> (is it 1000 Base T?)

per https://en.wikipedia.org/wiki/Fast_Ethernet#Nomenclature :

> The 100 in the media type designation refers to the transmission speed of 100 Mbit/s, while the BASE refers to [baseband](https://en.wikipedia.org/wiki/Baseband) signaling. The letter following the dash (T or F) refers to the physical medium that carries the signal (twisted pair or fiber, respectively), while the last character (X, 4, etc.) refers to the [line code](https://en.wikipedia.org/wiki/Line_code) method used.

So, yeah, I think we're after 1000BASE-T here.

> Another question I have is what difference the Base 10/100 T jacks vs the base 1000 T jacks are. Like there are the right number of pins present (8 for the 4 twisted pairs) and they each went through magnetics is there a real difference in functionality? Or is having 8 pins the same as saying it's for Base 1000 T?

At a guess, the difference is manufacturing process & quality control: 1000BASE-T probably has tighter requirements, and not wanting to test for those and/or not passing those tests is how you end up getting classified as an 100BASE-T RJ-45 jack.

**Reply by sethp** [reply]
Just a couple of quick notes here:

1. External LEDs & magnetics ought to drop the price on the jack significantly, right? Even $2/jack is pretty steep, since that means the 4x ports cost more than the GSW chip itself.
2. I wonder if there's any savings to be had by grouping ports together? Like, is a 2x or 4x port single module cheaper than a 1x port?

and, in somewhat reference to my wild A/B testing idea yesterday:

3. Is this a good place to play with edge connectors? i.e. if we make modular jacks, can we plug them into different switch chips to avoid either buying a bunch of extra jacks or desoldering/resoldering them a bunch?

**Reply by dougli1sqrd** [reply]
> The RJ-45 jacks are through hole parts, right?

Yeah so a "tab up" jack installed "upside down" would be tab-direction invariant (lol) with a "tab down" jack installed on the same side as the chip.

I guess it doesn't matter if they're not both on the same side?

**Reply by sethp** [reply]
> I guess it doesn't matter if they're not both on the same side?

I don't think so for our manufacturing capabilities (hot plate + iron); I would guess that slamming the top of the PCB into a solder wave for the jacks would have some undesirable consequences (populated or not), but for a rough prototype it sounds like we can do whichever.

**Reply by sethp** [reply]
re: external LEDs & magetnics with grouped ports, looks like it's possible to get down to ~$1/port on [mouser](https://www.mouser.com/c/connectors/modular-connectors-ethernet-connectors/?configuration=1x2%7C~1x4%7C~2x1~~2x2&performance%20category=Cat%205%2C%20Cat%205e~~Cat%206a&usoc%20codes=RJ45&sort=pricing&rp=connectors%2Fmodular-connectors-ethernet-connectors%7C~Configuration%7C~Performance%20Category) with something like this [1x4 configuration](https://www.mouser.com/ProductDetail/Phoenix-Contact/1149851?qs=7MVldsJ5UayGu5Wx4lDmmQ%3D%3D) with no LEDs and magnetics, and then maybe one of these [little $0.78 buggers](https://www.mouser.com/ProductDetail/Bourns/SM453230-181N7Y?qs=81r%252BiQLm7BSCr1yHDp19SQ%3D%3D) per port (via ["signal transformers" mouser search](https://www.mouser.com/c/passive-components/audio-transformers-signal-transformers/?type=1000Base-T%2C%2010GBase-T~~1G%20Base-T%20PoE%20Transformer%7C~Ethernet%20Transformers%7C~LAN~~LAN%20Discrete%20Modules%7C~LAN%20Transformer~~LAN%20Transformer%20Module&rp=passive-components%2Faudio-transformers-signal-transformers%7C~Type&sort=pricing) but I didn't put too much effort into it [there's probably parts that bake in multiple channels, and are therefore cheaper in aggregate]).

Looks like both parts have footprints available for download, too, so that's nice.

**Reply by dougli1sqrd** [reply]
>  if we make modular jacks

you can make modular jacks?? I guess it makes sense they would call it modular

Yeah I searched around previously for external transformers before and didn't know what I was looking for so I opted to stay with mag jacks. But maybe it could end up being cheaper separately!

**Reply by sethp** [reply]
Yeah, I'm not entirely sure what we're looking at still either: the turn ratio being 1:1 means we're not actually doing any "transforming", I think, just filtering ("choking") and DC isolation.

**Reply by sethp** [reply]
oh, I wanted to throw out there that for "modular jacks" what I meant was like, a common edge connector that we could plug different jack configurations into. Like, given some abstract controller that has 5Gb of aggregate bidirectional bandwidth, make it possible to plug in either 2x2.5GbE ports or 5x1GbE ports.

Sort of similar the framework model of having a common "bus" (for them, USB3.1; for us.... PCIe?) and a bunch of "port configurations" that hang off that bus. Given how spendy and tradeoff-laden the PHY layer seems to be, I wonder if adding more flexibility there would make sense.

I'm not thinking this is a "next" thing, or even a "next next," just something I'm curious about in general.

**Reply by dougli1sqrd** [reply]
ah yeah confusingly digikey calls the category that RJ45 jacks are in "modular/ethernet connectors"

**Comment by dougli1sqrd** [comment]
## Different Arrangements of the GSW141 Breakout PCB?

### RJ45 Jacks

* Internal Magnetics | External Magnetics
* LEDs | No LEDs
* 1000 (or higher) Base-T
    * I think 1G capable means that there are 4 twisted pairs
* "Tab Up" | "Tab Down"
    * This means the tab on the cable. When tab is down, the cat-5 pins all in a row point "up",
and vise versa when tab is up.
    * The GSW141 chip routes traces best with "tab down"
    * "Tab up" + on the reverse side of the board is equivalent to "tab down"
* How many ports per part



### Internal Magnetics

Use `ARJM11C7-502-KB-EW2` on mouser
* Tab up, so would have to be placed on the reverse side of the PCB
* $3.09 @ 10

4x 1 unit = $12.36


--------------------------------------------------------------------------------------

Use `TMJG26945AENL` on mouser, 1x2 port
* Tab down
* $11.34 per unit

2x 1 unit = $22.68


--------------------------------------------------------------------------------------

Use `TMJG46945AENL` on mouser, 1x4 port
* Tab down
* $18.84 per unit


### External Magnetics

* Mouser `1149848` 4 port RJ45 w/ LEDs, $4.34 each
    * Tab up, so would have to be placed on PCB reverse side
* [Mouser `LAN2VSOD48351C2`](https://www.mouser.com/ProductDetail/Bussmann-Eaton/LAN2VSOD48351C2?qs=QpmGXVUTftHO8SxsptDqJQ%3D%3D) dual port magnetics chip, ~$2 each

1 @ 4.34 + 2 @ $2 = $8.34

--------------------------------------------------------------------------------------

* Mouser `1149854` 2 port RJ45 w/LEDs, $2.78 each
    * Tab up, reverse PCB side
* Mouser `LAN2VSOD48351C2` dual port magnetics chip, ~$2 each

2 @ 2.78 + 2 @ 2 = $9.56
